components/Chips/tests/snapshot/MDCChipViewCustomTraitCollectionSnapshotTests.m

```diff
@@ -0,0 +1,121 @@
+  return self.traitCollectionOverride ?: [super traitCollection];
+}
+@end
+
+/**
+ A Snapshot test case for testing MDCChipViewWithCustomTraitCollection   the @c traitCollection
+ property.
+ */
+@interface MDCChipViewCustomTraitCollectionSnapshotTests : MDCSnapshotTestCase
+
+@property(nonatomic, strong) MDCContainerScheme *containerScheme;
+@property(nonatomic, strong) MDCChipViewWithCustomTraitCollection *chip;
+
+@end
+
+@implementation MDCChipViewCustomTraitCollectionSnapshotTests
+
+- (void)setUp {
+  [super setUp];
+
+  // Uncomment below to recreate all the goldens (or add the following line to the specific
+  // test you wish to recreate the golden for).
+  //          self.recordMode = YES;
+
+  self.chip = [[MDCChipViewWithCustomTraitCollection alloc] initWithFrame:CGRectMake(0, 0, 80, 80)];
+  self.containerScheme = [[MDCContainerScheme alloc] init];
+  self.containerScheme.colorScheme =
+      [[MDCSemanticColorScheme alloc] initWithDefaults:MDCColorSchemeDefaultsMaterial201804];
+}
+
+- (void)tearDown {
+  self.chip = nil;
+  self.containerScheme = nil;
+
+  [super tearDown];
+}
+
+#pragma mark - Helpers
+
+- (void)generateSnapshotAndVerifyForView:(UIView *)view {
+  UIView *snapshotView = [view mdc_addToBackgroundView];
+  [self snapshotVerifyView:snapshotView];
+}
+
+- (void)testDynamicColorSupport {
+#if defined(__IPHONE_13_0) && (__IPHONE_OS_VERSION_MAX_ALLOWED >= __IPHONE_13_0)
+  if (@available(iOS 13.0, *)) {
+    // Given
+    UIColor *dynamicShadowColor =
+        [UIColor colorWithDynamicProvider:^(UITraitCollection *traitCollection) {
+          if (traitCollection.userInterfaceStyle == UIUserInterfaceStyleLight) {
+            return UIColor.magentaColor;
+          } else {
+            return UIColor.greenColor;
+          }
+        }];
+    UIColor *dynamicBackgroundColor =
+        [UIColor colorWithDynamicProvider:^(UITraitCollection *traitCollection) {
+          if (traitCollection.userInterfaceStyle == UIUserInterfaceStyleLight) {
+            return UIColor.orangeColor;
+          } else {
+            return UIColor.redColor;
+          }
+        }];
+    UIColor *dynamicBorderColor =
+        [UIColor colorWithDynamicProvider:^(UITraitCollection *traitCollection) {
+          if (traitCollection.userInterfaceStyle == UIUserInterfaceStyleLight) {
+            return UIColor.yellowColor;
+          } else {
+            return UIColor.brownColor;
+          }
+        }];
+
+    [self.chip setBorderWidth:3 forState:UIControlStateHighlighted];
+    [self.chip setShadowColor:dynamicShadowColor forState:UIControlStateHighlighted];
+    [self.chip setBorderColor:dynamicBorderColor forState:UIControlStateHighlighted];
+    [self.chip setBackgroundColor:dynamicBackgroundColor forState:UIControlStateHighlighted];
+
+    // When
+    self.chip.traitCollectionOverride =
+        [UITraitCollection traitCollectionWithUserInterfaceStyle:UIUserInterfaceStyleDark];
+
+    self.chip.highlighted = YES;
```

Comment:
Tip: Would you please put the `highlighted` state change in the "Given"? That way we don't accidentally rely on `setHighlighted` to perform the color update and we can instead rely on `layoutSubviews` more directly.